src/chai/ArrayManager.cpp

```diff
@@ -33,10 +33,15 @@ ArrayManager::ArrayManager() :
 {
   m_pointer_map.clear();
   m_current_execution_space = NONE;
+  m_current_resource = nullptr;
   m_default_allocation_space = CPU;
 
   m_allocators[CPU] =
+#if defined(CHAI_ENABLE_CUDA) || defined(CHAI_ENABLE_HIP) || defined(CHAI_ENABLE_GPU_SIMULATION_MODE)
```

Comment:
I don't think we want this change. We're already pushing the limits of GPU memory, so we should just use host memory by default. There is an API for setting the allocator if someone wants pinned memory.

Reply:
If host memory is not pinned, the async copies won't be async. I agree that we can change the default back, but need to make sure this is effectively communicated to users.